### PR TITLE
Rename STOCKOUT to RESOURCE_POOL_EXHAUSTED

### PR DESCRIPTION
**Pull request by losipiuk**
We came into conclusion that using STOCKOUT as error code is too
specific. Migrating to more general term RESOURCE_POOL_EXHAUSTED.

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go [hunk 1]
```diff
@@ -40,8 +40,8 @@ const (
 	// ErrorCodeQuotaExceeded is error code used in InstanceErrorInfo if quota exceeded error occurs.
 	ErrorCodeQuotaExceeded = "QUOTA_EXCEEDED"
 
-	// ErrorCodeStockout is error code used in InstanceErrorInfo if stockout occurs.
-	ErrorCodeStockout = "STOCKOUT"
+	// ErrorCodeResourcePoolExhausted is error code used in InstanceErrorInfo if stockout occurs.
```
Comment (MaciekPytel) [inline cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go]: Update docstring too?

**Comment by MaciekPytel** [comment]
/approve
/lgtm
/hold 
Hold to address a single comment.

File: cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go (hunk 1 above)
Comment (losipiuk) [reply cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go]: Done

**Pull request merged**